CairoMakie/src/utils.jl

```diff
@@ -190,245 +190,232 @@ function clip2screen(p, res)
     return res .* s
 end
 
-@generated function project_line_points(scene, plot::T, positions, colors, linewidths) where {T <: Union{Lines, LineSegments}}
+
+
+function project_line_points(scene, plot::T, positions, colors, linewidths) where {T <: Union{Lines, LineSegments}}
     # If colors are defined per point they need to be interpolated like positions
     # at clip planes
-    per_point_colors = colors <: AbstractArray
-    per_point_linewidths = (T <: Lines) && (linewidths <: AbstractArray)
-    
-    quote
-        @get_attribute(plot, (space, model))
-
-        # Standard transform from input space to clip space
-        points = Makie.apply_transform(transform_func(plot), positions, space)
-        f32convert = Makie.f32_convert_matrix(scene.float32convert, space)
-        transform = Makie.space_to_clip(scene.camera, space) * model * f32convert
-        clip_points = Vector{Vec4f}(undef, length(points))
-        @inbounds for (i, point) in enumerate(points)
-            clip_points[i] = transform * to_ndim(Vec4d, to_ndim(Vec3d, point, 0), 1)
-        end
-        
-        # yflip and clip -> screen/pixel coords
-        res = scene.camera.resolution[]
+    per_point_colors = colors isa AbstractArray
+    per_point_linewidths = (T <: Lines) && (linewidths isa AbstractArray)
 
-        # clip planes in clip space
-        clip_planes = if Makie.is_data_space(space)
-            Makie.to_clip_space(scene.camera.projectionview[], plot.clip_planes[])
-        else
-            Makie.Plane3f[]
-        end
+    space = (plot.space[])::Symbol
+    model = (plot.model[])::Mat4d
+    # Standard transform from input space to clip space
+    points = Makie.apply_transform(transform_func(plot), positions, space)::typeof(positions)
```

Comment:
That's not guaranteed for 2D to 3D or 3D to 2D transformations...is this type annotation necessary here?  Can we specify that it will be a Vector of VecTypes maybe?

Reply:
This is required right now, since somehow it doesn't get inferred... This happens because transform_func doesn't infer. not sure what to do about it, I guess we could also put it behind a function barrier and extract transform_func before this function call and pass it as an argument.

Reply:
easiest would be to put everything after this behind another function barrier, `apply_transform` is already the function barrier for the dynamic `transform_func` result, but the stuff after is not guarded without `::typeof(positions)`